### PR TITLE
task-7

### PR DESCRIPTION
**Pull request by NurbekAman**
- [x] 1 authorization-service is added to the repo, has correct basicAuthorizer lambda and correct serverless.yaml file

- [x] 3 - import-service serverless.yaml file has authorizer configuration for the importProductsFile lambda. Request to the importProductsFile lambda should work only with correct authorization_token being decoded and checked by basicAuthorizer lambda. Response should be in 403 HTTP status if access is denied for this user (invalid authorization_token) and in 401 HTTP status if Authorization header is not provided.

- [x] 1 integration with front-end

totally 5

front lint to test https://d1aekbebeakto3.cloudfront.net/admin/products

authorization: bnVyYmVrdGVzdD1URVNUX1BBU1NPV1JE

example of csv file
image

### CODE REVIEW

**Comment by IvanPresmytsky** [comment]
@NurbekAman could you please add a PR description and related links for testing?

File: authorization-service/.env [hunk 1]
```diff
@@ -0,0 +1 @@
+nurbektest=TEST_PASSWORD
```
Comment (IvanPresmytsky) [inline authorization-service/.env:1]: this file should be added to .gitignore

File: authorization-service/handlers/basicAuthorization.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    var authorizationToken = event.authorizationToken;
+    const encodedCreds = authorizationToken.split(' ')[1];
+    const buff = Buffer.from(encodedCreds, 'base64');
+    const plainCreds = buff.toString('utf-8').split(':')[0].split('=');
+    const username = plainCreds[0];
+    const password = plainCreds[1];
+    console.log('creads', plainCreds);
+    const storedUserPassword = process.env[username];
+    const { nurbektest } = process.env;
+    console.log('username in env', nurbektest)
```
Comment (IvanPresmytsky) [inline authorization-service/handlers/basicAuthorization.ts]: Seems you forgot to remove this console

File: authorization-service/services/formatResponse.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  format(data: any, status: number) {
+    return {
+      headers: this.headers,
+      body: JSON.stringify(data),
+      statusCode: status
+    };
+  }
+
+  success(body: any | null) {
```
Comment (IvanPresmytsky) [inline authorization-service/services/formatResponse.ts:29]: You can leave just 'any' here. Any type is already included null.

File: import-service/serverless.ts [hunk 4]
```diff
@@ -116,6 +126,32 @@ const serverlessConfiguration: Serverless = {
...
+      GatewayResponseUnauthorized: {
+        Type: 'AWS::ApiGateway::GatewayResponse',
+        Properties: {
+          RestApiId: {
+            Ref: 'ApiGatewayRestApi'
+          },
+          ResponseType: 'UNAUTHORIZED',
+          ResponseParameters: {
+            'gatewayresponse.header.Access-Control-Allow-Origin': "'*'",
+            'gatewayresponse.header.Access-Control-Allow-Headers': "'*'",
```
Comment (IvanPresmytsky) [inline import-service/serverless.ts]: You can move headers to separate variable to avoid duplication

**Comment by NurbekAman** [comment]
> @NurbekAman could you please add a PR description and related links for testing?

I have a problem with cors. With postman it works, but by the web site is it not working. Give me a time to fix it)

**Comment by NurbekAman** [comment]
> @NurbekAman could you please add a PR description and related links for testing?

 Totally overcome cors and it seems front integration works)

**Comment by IvanPresmytsky** [comment]
> > @NurbekAman could you please add a PR description and related links for testing?
> 
> Totally overcome cors and it seems front integration works)

For now, there are no problems with CORS, but I managed to upload a file without adding any token to localStorage.

**Pull request merged**